scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -945,22 +951,19 @@
   ;; TODO these limits should come from the largest observed host from Fenzo
   ;; .getResourceStatus on TaskScheduler will give a map of hosts to resources; we can compute the max over those
   [{max-memory-gb :memory-gb max-cpus :cpus} offensive-jobs-ch jobs]
-  (let [max-memory-mb (* 1024.0 max-memory-gb)
-        categorized-jobs (group-by (fn [job]
-                                     (let [{memory-mb :mem
-                                            cpus :cpus} (util/job-ent->resources job)]
-                                       (if (or (> memory-mb max-memory-mb)
-                                               (> cpus max-cpus))
-                                         :offensive
-                                         :inoffensive)))
-                                   jobs)]
-    ;; Put offensive jobs asynchronically such that it could return the
-    ;; inoffensive jobs immediately.
-    (async/go
-      (when (seq (:offensive categorized-jobs))
-        (log/info "Found" (count (:offensive categorized-jobs)) "offensive jobs")
-        (async/>! offensive-jobs-ch (:offensive categorized-jobs))))
-    (:inoffensive categorized-jobs)))
+  (timers/time!
+    filter-offensive-jobs-duration
+    (let [max-memory-mb (* 1024.0 max-memory-gb)
+          is-offensive? (partial is-offensive? max-memory-mb max-cpus)
+          inoffensive (remove is-offensive? jobs)
+          offensive (filter is-offensive? jobs)]
+      ;; Put offensive jobs asynchronically such that it could return the
```

Comment:
My issue with partition-by is that I don't know which partition is which. Is there a principled way to determine that?